pkg/agent/openflow/pipeline.go

```diff
@@ -1337,16 +1338,19 @@ func (c *client) l3FwdServiceDefaultFlowsViaGW(ipProto binding.Protocol, categor
 	gatewayMAC := c.nodeConfig.GatewayConfig.MAC
 
 	flows := []binding.Flow{
-		// This flow is used to match the packets of Service traffic:
-		//	- NodePort/LoadBalancer request packets which pass through Antrea gateway and the Service Endpoint is not on
+		// This generates the default flow to match Service packets, traffic cases include:
+		//  - When Egress is not enabled:
+		//    - Service request packets sourced from Antrea gateway and destined for external Endpoint.
+		//	  - Service request packets sourced from local Pods and destined for external Endpoint.
+		//    - Service response packets sourced from local Pods.
 		//    local Pod CIDR or any remote Pod CIDRs.
-		//	- ClusterIP request packets which are from Antrea gateway and the Service Endpoint is not on local Pod CIDR
-		// 	or any remote Pod CIDRs.
-		//  - NodePort/LoadBalancer/ClusterIP response packets.
-		// The matched packets should leave through Antrea gateway, however, they also enter through Antrea gateway. This
-		// is hairpin traffic.
+		//	- When Egress is enabled:
+		//    - Service request packets sourced from Antrea gateway and destined for external Endpoint.
+		//    - Service response packets sourced from local Pods.
+		// Note that, when Egress is enabled, Service request packets sourced from local Pods and destined for external
+		// Endpoint should not be matched and such packets should be matched by Egress flow in L3ForwardingTable.
```

Comment:
I think the comment makes sense but the code change doesn't seem reflecting it. How does the flow skip "Service request packets sourced from local Pods"?

Reply:
I added a new priority `priorityDefault 180` and changed this flow's priority to 180. The priority of the flow that matches the Egress packets is still 190, as a result, Service request packets sourced from local Pods will be matched by the flow with priority 190.

Reply:
I was confused by `priorityDefault`, thinking it as a priority higher than `priorityLow`. And I still think the name is confusing with `priorityNormal`.
Anyway, while I was reviewing the flows in L3Forwarding, I saw there is already a flow which should ensure reply packets wouldn't even go to SNAT table: https://github.com/antrea-io/antrea/blob/ceae20622d2ccdc46dbf1958793c2459532e56f1/pkg/agent/openflow/pipeline.go#L1256-L1264
I think it was because `FromGatewayCTMark`.
I think we should look at why this flow doesn't take effect.

Reply:
@hongliangl you might miss this comment. I think the flows has considered this case but somehow doesn't work as expected

Reply:
Then should we ensure the packet have correct fromGateway mark regardless of whether it is for service, which can fix the issue by making the original design work as expected?

Reply:
Fixed.